### DISCUSSION TITLE
Is it possible to throttle the callback on a topic subscription?

### DISCUSSION THREAD

**Discussion by PierreSachot** [RobotWebTools/rclnodejs#936] [Q&A] [answered]
Hi,
I am currently working on an HMI which listen high frequency topics. HMI does not required to listen all messages, is it possible to call the createSubscription and ask for rclnodejs to drop messages?
Currently I drop messages inside my callback if I receive a new one shortly (with a timer which is really ugly). But even by doing this, rclnodejs is triggered at each messages and convert them from topics to TS (I suppose) which makes my backend CPU usage very high. 

Regards,
Pierre

**Comment by minggangw** [comment]
Hi @PierreSachot  does the message filter meet your requirement? You can find the example https://github.com/RobotWebTools/rclnodejs/blob/develop/example/subscription-content-filter-example.js

> rclnodejs is triggered at each messages and convert them from topics to TS (I suppose) which makes my backend CPU usage very high.

Yes, the `rclnodejs` constructs JS objects from the memory block which can take more time and CPU cost.

**Comment by PierreSachot** [comment, marked as answer]
Hi @minggangw,
Thanks a lot for your answer, you made me found the solution. I tried to filter on the timestamp which is in my topic header, but it does not work.

My solution was to set the HMI ros node frequency at a lower level in order to reduce the entire processing. 

Here is the solution (inside the node creation: 
`
const rate = YOUR_NODE.createRate(YOUR_DESIRED_FREQUENCY);
while(true) {
    rate.sleep();
    node.spin_once();
}
`

**Reply by minggangw** [reply]
Glad to hear your problem solved!
